### PR TITLE
basic typescript config

### PR DESCRIPTION
**Pull request by krzkaczor**

### CODE REVIEW

File: typescript/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "compilerOptions": {
+    "sourceMap": true,
+    "declaration": true,
+    "module": "commonjs",
+    "moduleResolution": "node",
+    "emitDecoratorMetadata": true,
```
Comment (holgerd77) [inline typescript/tsconfig.json]: I think we generally need ``"resolveJsonModule": true``, we have this use case on various libraries.

Comment (krzkaczor) [reply typescript/tsconfig.json]: Hmm yeah, this seems useful. I will add it 👍

Comment (holgerd77) [reply typescript/tsconfig.json]: Thanks!

Comment (krzkaczor) [reply typescript/tsconfig.json]: DONE

**Pull request merged**